tests/protocol/test_hostplugin.py

```diff
@@ -536,6 +538,50 @@ def test_validate_page_blobs(self):
                         test_goal_state,
                         exp_method, exp_url, exp_data)
 
+    def test_validate_http_request_when_uploading_logs(self):
+        def http_put_handler(url, *args, **kwargs):
+            if self.is_host_plugin_put_logs_request(url):
+                http_put_handler.args, http_put_handler.kwargs = args, kwargs
+                return MockResponse(body=b'', status_code=200)
+            self.fail('The upload logs request was sent to the wrong url: {0}'.format(url))
```

Comment:
the mock handler should handle the requests it is interested in and let the others go thru (instead of failing the test). this allows for cases in which, as the code evolves, additional requests are done. the mock protocol would raise if nobody handles the request